src/routes/transactions/helpers/kiln-native-staking.helper.ts

```diff
@@ -104,6 +111,92 @@ export class KilnNativeStakingHelper {
       data: args.transaction.data,
     };
   }
+
+  /**
+   * Gets the net value (staked + rewards) to withdraw from the native staking deployment
+   * based on the length of the publicKeys field in the transaction data.
+   *
+   * Note: this can only be used with `validatorsExit` or `batchWithdrawCLFee` transactions
+   * as the have `_publicKeys` field in the decoded data.
+   *
+   * Each {@link KilnDecoder.KilnPublicKeyLength} characters represent a validator to withdraw,
+   * and each native staking validator has a fixed amount of 32 ETH to withdraw.
+   *
+   * @param dataDecoded - the decoded data of the transaction
+   * @param chainId - the ID of the chain where the native staking deployment lives
+   * @param safeAddress - the Safe staking
+   * @returns the net value to withdraw from the native staking deployment
+   */
+  public async getValueFromDataDecoded(args: {
+    chainId: string;
+    safeAddress: `0x${string}`;
+    dataDecoded: DataDecoded;
+  }): Promise<number> {
+    const publicKeys = this.getPublicKeysFromDataDecoded(args.dataDecoded);
+    if (publicKeys.length === 0) {
+      return 0;
+    }
+    const stakes = await this.stakingRepository.getStakes({
+      chainId: args.chainId,
+      safeAddress: args.safeAddress,
+      validatorsPublicKeys: publicKeys,
+    });
+    return stakes.reduce((acc, stake) => {
+      const netValue = stake.net_claimable_consensus_rewards ?? '0';
+      return acc + Number(netValue);
+    }, 0);
+  }
+
+  /**
+   * Gets public keys from decoded `requestValidatorsExit` or `batchWithdrawCLFee` transactions
+   * @param dataDecoded - the transaction decoded data.
+   * @returns the public keys from the transaction decoded data.
+   */
+  public getPublicKeysFromDataDecoded(
+    dataDecoded: DataDecoded,
+  ): Array<`0x${string}`> {
+    if (
+      !['requestValidatorsExit', 'batchWithdrawCLFee'].includes(
+        dataDecoded.method,
+      )
+    ) {
+      throw new UnprocessableEntityException(
+        `${dataDecoded.method} does not contain _publicKeys`,
+      );
+    }
```

Comment:
Although we have a comment regarding this, I added the extra check as it threw me off a few times.